### PR TITLE
feat (systemaddom): #3148 make the sections (+Top Sites) collapsible

### LINKED ISSUE #3148: Allow collapsible sections that can be collapsed
Matching the Test Pilot behavior to shrink a section to just the header

### PR DESCRIPTION
**Pull request by rlr**
Once again, a lot of the code changes is moving stuff around.

Fixes #3148

### CODE REVIEW

**Comment by Mardak** [comment]
We don't want to merge things not-for-57 yet.

File: system-addon/content-src/styles/_icons.scss [hunk 1]
```diff
@@ -104,4 +104,19 @@
...
+    background-size: $smaller-icon-size;
+    height: $smaller-icon-size;
+    width: $smaller-icon-size;
+  }
+
+  &.icon-arrowhead-forward {
+    background-image: url('#{$image-path}glyph-arrowhead-down-12.svg');
+    background-size: $smaller-icon-size;
+    height: $smaller-icon-size;
+    transform: rotate(270deg);
```
Comment (Mardak) [inline system-addon/content-src/styles/_icons.scss]: Probably need 90deg for rtl?

**Comment by rlr** [comment]
@csadilek I've addressed the impressions issue here 9d28eae56b980d89ddfcd2383322d341c695315e

**Comment by csadilek** [comment]
@rlr looking good. Found one issue when testing the update: After collapsing the section, I get one more view impression ping when opening a new tab, subsequent new tabs are fine. Reopening/Expanding the section works fine too! Turned ping-centre.log on to see this. Happy to help debug this.

**Comment by csadilek** [comment]
@rlr I've just verified that this one additional ping happens after any pref change. So, it's unrelated to your changes. I will take a look and file a follow-up issue, if needed.

**Comment by as-pine-proxy** [comment]
[Treeherder: 15fbf909aa4907b551120b206f17e66fdc87aae7](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=15fbf909aa4907b551120b206f17e66fdc87aae7)

**Pull request merged**